src/NUnitTestExtractor/NUnitTestExtractorTests/TestExtractorTests.cs

```diff
@@ -0,0 +1,255 @@
+
+                switch (myLevel)
+                {
+                    case NUnitTestExtractorApp.Level.Namespace:
+                        regex = new Regex(@"\w+\s\|\s[^\.]");
+                        break;
+
+                    case NUnitTestExtractorApp.Level.Class:
+                        regex = new Regex(@"\w+\s\|\s[^\.\.^\.]");
+                        break;
+
+                    case NUnitTestExtractorApp.Level.Function:
+                        regex = new Regex(@"\w+\s\|\s[^\.\.^\.^\.]");
+                        break;
+                }
+
+                Assert.That(regex.IsMatch(line));
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic")]
+        [TestCaseSource(typeof(InvalidDataTestCasesCollection))]
+        [Test]
+        public void WritingToStdout_InvalidDllAndEveryLevel_NothingWritten(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            using (Process process = SetUpProcess(dll, level))
+            {
+                Assert.That(process.StandardOutput.ReadToEnd(), Is.Empty);
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic")]
+        [TestCaseSource(typeof(ValidDataTestCasesCollection))]
+        [Test]
+        public void WritingToStdout_ValidDllAndEveryLevel_ContainsNoDuplicates(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            using (Process process = SetUpProcess(dll, level))
+            {
+                string line = string.Empty;
+
+                List<string> lines = new List<string>();
+
+                while ((line = process.StandardOutput.ReadLine()) != null)
+                {
+                    lines.Add(line);
+                }
+
+                Assert.AreEqual(lines.Count, lines.Distinct().Count());
+            } 
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic")]
+        [TestCase("NAMESPACE")]
+        [TestCase("namespace")]
+        [TestCase("NaMeSpAcE")]
+        [Test]
+        public void ParsingLevel_NamespaceWrittenInDifferentCases_WillParseWithoutError(string level)
+        {
+            Assert.That(NUnitTestExtractorApp.ParseLevel(level), Is.EqualTo(NUnitTestExtractorApp.Level.Namespace));
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic")]
+        [TestCase("CLASS")]
+        [TestCase("class")]
+        [TestCase("ClAsS")]
+        [Test]
+        public void ParsingLevel_ClassWrittenInDifferentCases_WillParseWithoutError(string level)
+        {
+            Assert.AreEqual(NUnitTestExtractorApp.ParseLevel(level), (NUnitTestExtractorApp.Level.Class));
```

Comment:
For the Assert.AreEqual() function, the first parameter should be the expected value and the next should be the actual value, otherwise the error message it produces will be backwards (if you temporarily change this to AreNotEqual & run the test, you'll see what I mean).